packages/core/src/postProcess/PostProcessManager.ts

```diff
@@ -1,73 +1,352 @@
+import { Vector3 } from "@galacean/engine-math";
+import { Camera } from "../Camera";
 import { PipelineUtils } from "../RenderPipeline/PipelineUtils";
-import { RenderContext } from "../RenderPipeline/RenderContext";
 import { Scene } from "../Scene";
+import { Logger } from "../base/Logger";
 import { Material } from "../material";
-import { Shader } from "../shader";
-import { RenderTarget, Texture2D } from "../texture";
-import { BloomEffect, TonemappingEffect } from "./effects";
+import { Collider, ColliderShape } from "../physics";
+import { RenderTarget, Texture2D, TextureFilterMode, TextureFormat, TextureWrapMode } from "../texture";
+import { PostProcess } from "./PostProcess";
+import { PostProcessEffect } from "./PostProcessEffect";
+import { PostProcessPass } from "./PostProcessPass";
 
 /**
- * @internal
+ * A global manager of the PostProcess.
  */
-export class _PostProcessManager {
-  static readonly UBER_SHADER_NAME = "UberPost";
-
-  /**
-   * Whether the post process manager is active.
-   */
-  isActive = true;
-
+export class PostProcessManager {
   /** @internal */
   _uberMaterial: Material;
   /** @internal */
-  _bloomEffect: BloomEffect;
+  _postProcessNeedSorting = false;
+  /** @internal */
+  _postProcessPassNeedSorting = false;
   /** @internal */
-  _tonemappingEffect: TonemappingEffect;
+  _activeStateChangeFlag = false;
+
+  private _postProcessPasses: PostProcessPass[] = [];
+  private _activePostProcesses: PostProcess[] = [];
+  private _activePostProcessPasses: PostProcessPass[] = [];
+  private _isActive = false;
+  private _swapRenderTarget: RenderTarget;
+  private _srcRenderTarget: RenderTarget;
+  private _destRenderTarget: RenderTarget;
+  private _currentSourceRenderTarget: RenderTarget;
+  private _currentDestRenderTarget: RenderTarget;
+  private _blendEffectMap = new Map<typeof PostProcessEffect, PostProcessEffect>();
+  private _defaultEffectMap = new Map<typeof PostProcessEffect, PostProcessEffect>();
+  private _remainActivePassCount = 0;
+  private _tempColliders: Collider[] = [];
+  private _tempColliderShapes: ColliderShape[] = [];
+  private _tempVector3 = new Vector3();
+
+  /**
+   * Whether has any active pass and active effect.
+   */
+  get isActive(): boolean {
```

Comment:
It looks more like an internal flag to determine whether an effect needs to be produced. isActive is more like a switch expression of itself. Do not use isActive, and you should consider whether to make it public.